### ISSUE TITLE
[a11y]: various issues in ai-chat feedback form

### ISSUE THREAD

**Issue by Daniel-Schulz-Private** [carbon-design-system/carbon-labs#379]
### Browser

Firefox

### Operating System

Windows

### Package version

0.14.1-canary.347

### React version

^1.66.0

### Automated testing tool and ruleset

none

### Assistive technology

JAWS 2025

### Description

Here are some problems with the feedback form:
1. JAWS does not announce that a new window opened after a thumb button was selected
2. The feedback window has no focus -> need to hit tab 2 times after a bad response and 3 times after a good response to reach the feedback window.
3. It does not read the feedback page - "why did you chose this rating" - it jumps directly to the buttons. It also does not read the text "To better understand your feedback...", but jumps right to the checkbox
4. After selecting the checkbox, the next item we reach with tab is the "close" button. After that "submit" -> a blind person would probably hit "close" instead of "submit".
5. After submitting a feedback, it is not clear when tabbing through what the current status of the thumbs button is.

Details:
Here are the announcement read out loud when tabbing through the thumbs:
- "good response button collapsed, good response button, to activate press space bar"
-> press tab
- "bad response button collapsed, bad response button, to activate press space bar"
-> press space bar
- "remove rating, remove rating button, to activate press space bar" (the bad response button is now selected)
-> it does not say that a new window opened
-> how does a blind person know that the window can be reached by hitting tab 2 (bad response) or 3 (good response) times?
-> press tab again
- "Regenerate button"
-> Tab
- "Incorrect unselected - incorrect button" (reading the first feedback button)
-> Tab until I reach the checkbox (it does not read the text above the checkbox)
-> select the checkbox
-> Tab
- "Close" (this can also be seen without JAWS - a seeing person would also expect to tab to "submit" instead of "close")
-> Tab
- "Submit"
-> Hitting space bar to submit
-> window closes
-> now tabbing to the selected bad response thumb, it will read
- "remove rating button expanded, remove rating button, to activate press space bar"
-> the thumb is not expanded (the feedback menu is closed), but selected

### Reproduction/example

none

### Steps to reproduce

1. leverage the ai-chat 
2. select either the thumbs up or thumbs down icon in the chat history below any given chat response message

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-labs/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-labs/issues) for duplicate problems
